src/model/BaseModel.js

```diff
@@ -0,0 +1,85 @@
+import firebase from 'firebase';
+import { db } from '../services/FirebaseService';
+
+function timestamp() {
+  return firebase.firestore.FieldValue.serverTimestamp();
+}
+function create({ collection, data }) {
+  const [key, value] = Object.values(data);
+  return db
+    .collection(collection)
+    .doc(key)
+    .set({ ...value, dateUpdated: timestamp(), dateCreated: timestamp() });
+}
+
+function createOrUpdate({ collection, id, data }) {
+  return db.collection(collection).doc(id).set({ data });
+}
+
+function set({ collection, data, id }) {
+  return db
+    .collection(collection)
+    .doc(id)
+    .set({ ...data, dateUpdated: timestamp() });
+}
+
+async function update({ collection, data, id }) {
+  const [key, value] = Object.values(data);
```

Comment:
Keep in mind, `Object.values` returns **ONLY** the values of the object. There will be no keys in there, so this will probably fail when you invoke it with the data you want. I think what you are going for is `Object.entries` which returns a key value pair of the key and value. 

![Screen Shot 2021-08-07 at 3 50 03 PM](https://user-images.githubusercontent.com/32306822/128593052-efa980f9-51df-4d7b-a344-2ae076d5da2c.png)

Either way let's use Michael's suggestion for updating as it's more straightforward so we don't have to do any object destructuring.

![Screen Shot 2021-08-07 at 3 41 27 PM](https://user-images.githubusercontent.com/32306822/128592855-2bf4638d-f4c1-4843-9296-a26ac8282f2e.png)

Reply:
I don't quite understand what you mean, are you suggesting that the different models, SnippetModel, UserModel etc.. don't call BaseModel.create/update but instead have their own implementations?